.github/workflows/docs-deploy-surge.yml

```
@@ -0,0 +1,67 @@
+# Use this starter workflow to deploy HTML generated by Antora to surge.sh
+# Docs are published at <org>-<repo>-<deployid>.surge.sh
+# By default, this workflow runs on completion of a workflow called "Verify PR"
+# This workflow expects the triggering workflow to generate an artifact called "docs"
+
+# - update the reference to "docs" and "docs.zip" in this workflow if your triggering workflow generates an artifact with a different name
+name: "Deploy to surge"
+
+on:
+  workflow_run:
+    workflows: ["Verify PR"]
+    types:
+      - completed
+
+jobs:
+  publish-docs:
+    # if: github.event.workflow_run.conclusion == 'success'
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: "Download built documentation"
+        uses: actions/github-script@v6.0.0
+        with:
+          script: |
+            var artifacts = await github.rest.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: ${{ github.event.workflow_run.id }},
+            });
+            var matchArtifact = artifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "docs"
+            })[0];
+            var download = await github.rest.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifact.id,
+               archive_format: 'zip',
+            });
+            var fs = require('fs');
+            fs.writeFileSync('${{ github.workspace }}/docs.zip', Buffer.from(download.data));
+      
+      - run: unzip docs.zip
+      
+      - id: get-deploy-id
+        run: |
+          deployid=$(<deployid)
+          echo "::set-output name=deploy-id::$deployid"
```

Let's verify that the `deployid` (the PR number) is an integer. We had a security issue raised on the GraphQL repo regarding this so we should follow the same pattern here.
```suggestion
          deployid=$(<deployid)
          case "$deployid" in ''|*[!0-9]*) echo "Provided PR number is not an integer"; exit 1 ;; esac
          echo "::set-output name=deploy-id::$deployid"
```